controllers/crypto/productsController.js

```diff
@@ -0,0 +1,77 @@
+const logger = require('../../utils/logger')
+const productQuery = require('../../models/crypto/products')
+
+/**
+ * Fetches the data of products
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getProducts = async (req, res) => {
+  try {
+    const allProducts = await productQuery.fetchProducts()
+
+    return res.status(200).json({
+      message: 'Products returned successfully!',
+      Products: allProducts
+    })
+  } catch (error) {
+    logger.error(`Error while fetching all products: ${error}`)
+    return res.boom.badImplementation('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Fetches the data of product
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getProduct = async (req, res) => {
+  try {
+    const product = await productQuery.fetchProduct(req.params.id)
+    if (product) {
+      return res.status(200).json({
+        message: 'Product returned successfully!',
+        Products: product
+      })
+    }
+    return res.boom.notFound('Product not found!')
+  } catch (error) {
+    logger.error(`Error while fetching product: ${error}`)
+    return res.boom.badImplementation('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Add new Product
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - Product object
+ * @param res {Object} - Express response object
+ */
+const addNewProduct = async (req, res) => {
+  try {
+    const product = await productQuery.addProduct(req.body)
```

Comment:
Let us not directly send `body` to the database
If I just send `Hello Swaraj` in `body`, it'll be saved to the db directly which is not desired

We need to add validations in it. `/users` routes are using validations, please check them and proceed correspondingly